python/GafferOSLTest/OSLShaderTest.py

```diff
@@ -1285,5 +1288,26 @@ def testVdbVolumeType( self ) :
 
 		self.assertEqual( n["type"].getValue(), "osl:shader" )
 
+	def testLoadSurfaceWithOutputParameters( self ) :
+
+		script = Gaffer.ScriptNode()
+		script["fileName"].setValue( pathlib.Path( __file__ ).parent / "scripts" / "surfaceWithOutputs-1.3.9.0.gfr" )
+		script.load()
+
+		# The script was saved from a version of Gaffer where we didn't load output parameters
+		# for surface shaders. But we expect to load them now.
+
+		self.assertEqual( script["Constant"]["out"].keys(), [ "out" ] )
+		self.assertIsInstance( script["Constant"]["out"]["out"], GafferOSL.ClosurePlug )
+
+		# And we expect the old connection from `Constant.out` to continue to work, even though
+		# we would now expect an equivalent connection to be made from `Constant.out.out`.
```

Comment:
The only question I can think of for doing differently would be whether it would be worth adding a compatibility config for converting old connections to the new format, vs continuing to support a mix of old and new connections. I don't actually really have an opinion either way, just figured I'd mention it while I was thinking about it. This approach seems fine.

Reply:
I think it's quite hard to foresee all the possible consequences of such a config, so for now at least I'd like to avoid that.